overlord/ifacestate/handlers.go

```diff
@@ -744,6 +744,64 @@ func (m *InterfaceManager) doAutoConnect(task *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+// doDisconnectInterfaces creates tasks for disconnecting an interface and running its interface hooks.
+func (m *InterfaceManager) doDisconnectInterfaces(task *state.Task, _ *tomb.Tomb) error {
+	st := task.State()
```

Comment:
Question: should this function lock or assume the state is locked?

Reply:
it's a task handler, it's taking the lock as usual for those